services/web/server/src/simcore_service_webserver/exporter/formatters/formatter_v1.py

```diff
@@ -155,12 +157,15 @@ async def generate_directory_contents(
     await ProjectFile.model_to_file(root_dir=root_folder, **project_data)
 
 
+ETag = str
+
+
 async def upload_file_to_storage(
     app: web.Application,
     link_and_path: LinkAndPath2,
     user_id: int,
     session: ClientSession,
-) -> None:
+) -> Tuple[LinkAndPath2, ETag]:
```

Comment:
Feel free to add the `ETag` directly inside `LinkAndPath2` (if it makes sense) as it is used for downloading files.

Reply:
well the ETag only exists after the upload is successfully done.